### PR TITLE
scripts: add script for listing deprecations

### PR DESCRIPTION
**Pull request by Rugvip**
To help showing what deprecations exist, and when they were released

image

### CODE REVIEW

File: scripts/list-deprecations.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+const _ = require('lodash');
+const fs = require('fs-extra');
+const globby = require('globby');
+const { resolve: resolvePath, relative: relativePath } = require('path');
+const { execFile: execFileCb } = require('child_process');
+const { promisify } = require('util');
+
+const execFile = promisify(execFileCb);
+
+const WORKER_COUNT = 16;
```
Comment (freben) [inline scripts/list-deprecations.js:29]: isn't that pretty harsh :) for a disk heavy thing

File: scripts/list-deprecations.js [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    this.items = items;
+    this.index = 0;
+  }
+
+  push(...items) {
+    this.items.push(...items);
+  }
+
+  pop() {
+    const item = this.items[this.index];
```
Comment (freben) [inline scripts/list-deprecations.js:44]: can't this entire class be replaced with just `push()` and `shift()` on an array?

**Review by freben — APPROVED** [review]
Looks nice

File: scripts/list-deprecations.js (hunk 2 above)
Comment (Rugvip) [reply scripts/list-deprecations.js:44]: It used to make more sense but couldn't be arsed removing it xP

File: scripts/list-deprecations.js (hunk 1 above)
Comment (Rugvip) [reply scripts/list-deprecations.js:29]: 8 and 32 got similar performance on my machine, while going above or below that started causing issues, so felt this was a good middle ground

**Review by jhaals — APPROVED** [review]
:shipit:

**Pull request merged**